web/src/pm_list.ts

```diff
@@ -197,16 +205,35 @@ function zoom_in(): void {
     $(".direct-messages-container").removeClass("zoom-out").addClass("zoom-in");
     $("#streams_list").hide();
     $(".left-sidebar .right-sidebar-items").hide();
+    $("#left_sidebar_scroll_container .simplebar-content-wrapper").attr("tabindex", "-1");
+
+    const $filter = $(".direct-messages-list-filter").expectOne();
+    $filter.trigger("focus");
 }
 
 function zoom_out(): void {
     zoomed = false;
-    update_private_messages();
+    if (!clear_search()) {
+        update_private_messages();
+    }
     $(".direct-messages-container").removeClass("zoom-in").addClass("zoom-out");
     $("#streams_list").show();
     $(".left-sidebar .right-sidebar-items").show();
+    $("#left_sidebar_scroll_container .simplebar-content-wrapper").attr("tabindex", "0");
+}
+
+export function clear_search(): boolean {
+    const $filter = $(".direct-messages-list-filter").expectOne();
+    if ($filter.val() !== "") {
+        $filter.val("");
+        update_private_messages();
+        return true;
+    }
+    return false;
 }
```

Comment:
You should never have a function like this that returns `true/false` where the combination of name and comments don't explain what the return value means.

I think this returns "was there text to clear?", but it could also be semantically "Did this rerender the sidebar?", and I'm not sure which you intend, for example.

Reply:
Looking at how the parameter is used, to skip a second `update_private_messages` in `zoom_out`, maybe you just want a `force_rerender` parameter to this function, and no return value?

Reply:
Fixed!